File: LemonadeExchange/js/MuffinRushAssets.js
/**
 * Created by vedi on 1/28/14.
 */
function MuffinRushAssets() {

  const MUFFIN_CURRENCY_ITEM_ID = "currency_muffin";
  const TENMUFF_PACK_PRODUCT_ID = "android.test.refunded";
  const FIFTYMUFF_PACK_PRODUCT_ID = "android.test.canceled";
  const FOURHUNDMUFF_PACK_PRODUCT_ID = "android.test.purchased";
  const THOUSANDMUFF_PACK_PRODUCT_ID = "android.test.item_unavailable";
  const NO_ADDS_NONCONS_PRODUCT_ID = "no_ads";

  const MUFFINCAKE_ITEM_ID = "fruit_cake";
  const PAVLOVA_ITEM_ID = "pavlova";
  const CHOCLATECAKE_ITEM_ID = "chocolate_cake";
  const CREAMCUP_ITEM_ID = "cream_cup";


  /** Virtual Currencies **/
  var muffinCurrency = Soomla.Models.VirtualCurrency.create({
    name: "Muffins",
    description: "",
    itemId: MUFFIN_CURRENCY_ITEM_ID
  });


  /** Virtual Currency Packs **/
  var tenmuffPack = Soomla.Models.VirtualCurrencyPack.create({
    name: "10 Muffins",
    description: "Test refund of an item",
    itemId: "muffins_10",
    currency_amount: 10,
    currency_itemId: MUFFIN_CURRENCY_ITEM_ID,
    purchasableItem: Soomla.Models.PurchaseWithMarket.createWithMarketItem(TENMUFF_PACK_PRODUCT_ID, 0.99)
  });

  var fiftymuffPack = Soomla.Models.VirtualCurrencyPack.create({
    name: "50 Muffins",
    description: "Test cancellation of an item",
    itemId: "muffins_50",
    currency_amount: 50,
    currency_itemId: MUFFIN_CURRENCY_ITEM_ID,
    purchasableItem: Soomla.Models.PurchaseWithMarket.createWithMarketItem(FIFTYMUFF_PACK_PRODUCT_ID, 1.99)
  });

  var fourhundmuffPack = Soomla.Models.VirtualCurrencyPack.create({
    name: "400 Muffins",
    description: "Test purchase of an item",
    itemId: "muffins_400",
    currency_amount: 400,
    currency_itemId: MUFFIN_CURRENCY_ITEM_ID,
    purchasableItem: Soomla.Models.PurchaseWithMarket.createWithMarketItem(FOURHUNDMUFF_PACK_PRODUCT_ID, 4.99)
  });

  var thousandmuffPack = Soomla.Models.VirtualCurrencyPack.create({
    name: "1000 Muffins",
    description: "Test item unavailable",
    itemId: "muffins_1000",
    currency_amount: 1000,
    currency_itemId: MUFFIN_CURRENCY_ITEM_ID,
    purchasableItem: Soomla.Models.PurchaseWithMarket.createWithMarketItem(THOUSANDMUFF_PACK_PRODUCT_ID, 8.99)
  });

  /** Virtual Goods **/
  var muffincakeGood = Soomla.Models.SingleUseVG.create({
    name: "Fruit Cake",
    description: "Customers buy a double portion on each purchase of this cake",
    itemId: "fruit_cake",
    purchasableItem: Soomla.Models.PurchaseWithVirtualItem.create({
      pvi_itemId: MUFFIN_CURRENCY_ITEM_ID,
      pvi_amount: 225
    })
  });

  var pavlovaGood = Soomla.Models.SingleUseVG.create({
    name: "Pavlova",
    description: "Gives customers a sugar rush and they call their friends",
    itemId: "pavlova",
    purchasableItem: Soomla.Models.PurchaseWithVirtualItem.create({
      pvi_itemId: MUFFIN_CURRENCY_ITEM_ID,
      pvi_amount: 175
    })
  });

  var tenPavlovaGoods = Soomla.Models.SingleUsePackVG.create({
    good_itemId: "pavlova",
    good_amount: 10,
    name: "10 Pavlova",
    description: "Gives customers a sugar rush and they call their friends",
    itemId: "pavlova_10",
    purchasableItem: Soomla.Models.PurchaseWithVirtualItem.create({
      pvi_itemId: MUFFIN_CURRENCY_ITEM_ID,
      pvi_amount: 1750
    })
  });

  var choclatecakeGood = Soomla.Models.SingleUseVG.create({
    name: "Chocolate Cake",
    description: "A classic cake to maximize customer satisfaction",
    itemId: "chocolate_cake",
    purchasableItem: Soomla.Models.PurchaseWithVirtualItem.create({
      pvi_itemId: MUFFIN_CURRENCY_ITEM_ID,
      pvi_amount: 250
    })
  });

  var creamcupGood = Soomla.Models.SingleUseVG.create({
    name: "Cream Cup",
    description: "Increase bakery reputation with this original pastry",
    itemId: "cream_cup",
    purchasableItem: Soomla.Models.PurchaseWithVirtualItem.create({
      pvi_itemId: MUFFIN_CURRENCY_ITEM_ID,
      pvi_amount: 50
    })
  });

  var tenCreamcupGoods = Soomla.Models.SingleUsePackVG.create({
    good_itemId: "cream_cup",
    good_amount: 10,
    name: "10 Cream Cup",
    description: "Increase bakery reputation with this original pastry",
    itemId: "cream_cup_10",
    purchasableItem: Soomla.Models.PurchaseWithVirtualItem.create({
      pvi_itemId: MUFFIN_CURRENCY_ITEM_ID,
      pvi_amount: 500
    })
  });

  var showRoomGood = Soomla.Models.LifetimeVG.create({
    name: "Show Room",
    description: "Show Room ",
    itemId: "show_room",
    purchasableItem: Soomla.Models.PurchaseWithVirtualItem.create({
      pvi_itemId: MUFFIN_CURRENCY_ITEM_ID,
      pvi_amount: 100
    })
  });

  var showRoomGood0 = Soomla.Models.UpgradeVG.create({
    good_itemId: "show_room",
    prev_itemId: null,
    next_itemId: "show_room_1",
    name: "Show Room L0",
    description: "",
    itemId: "show_room_0",
    purchasableItem: Soomla.Models.PurchaseWithVirtualItem.create({
      pvi_itemId: MUFFIN_CURRENCY_ITEM_ID,
      pvi_amount: 220
    })
  });

  var showRoomGood1 = Soomla.Models.UpgradeVG.create({
    good_itemId: "show_room",
    prev_itemId: "show_room_0",
    next_itemId: "show_room_2",
    name: "Show Room L1",
    description: "",
    itemId: "show_room_1",
    purchasableItem: Soomla.Models.PurchaseWithVirtualItem.create({
      pvi_itemId: MUFFIN_CURRENCY_ITEM_ID,
      pvi_amount: 220
    })
  });

  var showRoomGood2 = Soomla.Models.UpgradeVG.create({
    good_itemId: "show_room",
    prev_itemId: "show_room_1",
    next_itemId: "show_room_3",
    name: "Show Room L2",
    description: "",
    itemId: "show_room_2",
    purchasableItem: Soomla.Models.PurchaseWithVirtualItem.create({
      pvi_itemId: MUFFIN_CURRENCY_ITEM_ID,
      pvi_amount: 220
    })
  });

  var showRoomGood3 = Soomla.Models.UpgradeVG.create({
    good_itemId: "show_room",
    prev_itemId: "show_room_2",
    next_itemId: "show_room_4",
    name: "Show Room L3",
    description: "",
    itemId: "show_room_3",
    purchasableItem: Soomla.Models.PurchaseWithVirtualItem.create({
      pvi_itemId: MUFFIN_CURRENCY_ITEM_ID,
      pvi_amount: 220
    })
  });

  var showRoomGood4 = Soomla.Models.UpgradeVG.create({
    good_itemId: "show_room",
    prev_itemId: "show_room_3",
    next_itemId: null,
    name: "Show Room L4",
    description: "",
    itemId: "show_room_4",
    purchasableItem: Soomla.Models.PurchaseWithVirtualItem.create({
      pvi_itemId: MUFFIN_CURRENCY_ITEM_ID,
      pvi_amount: 220
    })
  });

  var deliveryVehicleGood = Soomla.Models.LifetimeVG.create({
    name: "Delivery Vehicle",
    description: "Delivery Vehicle",
    itemId: "delivery_vehicle",
    purchasableItem: Soomla.Models.PurchaseWithVirtualItem.create({
      pvi_itemId: MUFFIN_CURRENCY_ITEM_ID,
      pvi_amount: 20
    })
  });

  var deliveryVehicleGood0 = Soomla.Models.UpgradeVG.create({
    good_itemId: "delivery_vehicle",
    prev_itemId: null,
    next_itemId: "delivery_vehicle_1",
    name: "Delivery Vehicle 0",
    description: "",
    itemId: "delivery_vehicle_0",
    purchasableItem: Soomla.Models.PurchaseWithVirtualItem.create({
      pvi_itemId: MUFFIN_CURRENCY_ITEM_ID,
      pvi_amount: 20
    })
  });

  var deliveryVehicleGood1 = Soomla.Models.UpgradeVG.create({
    good_itemId: "delivery_vehicle",
    prev_itemId: "delivery_vehicle_0",
    next_itemId: "delivery_vehicle_2",
    name: "Delivery Vehicle 1",
    description: "",
    itemId: "delivery_vehicle_1",
    purchasableItem: Soomla.Models.PurchaseWithVirtualItem.create({
      pvi_itemId: MUFFIN_CURRENCY_ITEM_ID,
      pvi_amount: 20
    })
  });

  var deliveryVehicleGood2 = Soomla.Models.UpgradeVG.create({
    good_itemId: "delivery_vehicle",
    prev_itemId: "delivery_vehicle_1",
    next_itemId: "delivery_vehicle_3",
    name: "Delivery Vehicle 2",
    description: "",
    itemId: "delivery_vehicle_2",
    purchasableItem: Soomla.Models.PurchaseWithVirtualItem.create({
      pvi_itemId: MUFFIN_CURRENCY_ITEM_ID,
      pvi_amount: 20
    })
  });

  var deliveryVehicleGood3 = Soomla.Models.UpgradeVG.create({
    good_itemId: "delivery_vehicle",
    prev_itemId: "delivery_vehicle_2",
    next_itemId: "delivery_vehicle_4",
    name: "Delivery Vehicle 3",
    description: "",
    itemId: "delivery_vehicle_3",
    purchasableItem: Soomla.Models.PurchaseWithVirtualItem.create({
      pvi_itemId: MUFFIN_CURRENCY_ITEM_ID,
      pvi_amount: 20
    })
  });

  var deliveryVehicleGood4 = Soomla.Models.UpgradeVG.create({
    good_itemId: "delivery_vehicle",
    prev_itemId: "delivery_vehicle_3",
    next_itemId: null,
    name: "Delivery Vehicle 4",
    description: "",
    itemId: "delivery_vehicle_4",
    purchasableItem: Soomla.Models.PurchaseWithVirtualItem.create({
      pvi_itemId: MUFFIN_CURRENCY_ITEM_ID,
      pvi_amount: 20
    })
  });

  var fatCatGood = Soomla.Models.EquippableVG.create({
    equipping: Soomla.Models.EquippableVG.EquippingModel.CATEGORY,
    name: "Fat Cat",
    description: "Fat cat description",
    itemId: "fat_cat",
    purchasableItem: Soomla.Models.PurchaseWithVirtualItem.create({
      pvi_itemId: MUFFIN_CURRENCY_ITEM_ID,
      pvi_amount: 27
    })
  });

  var happiHippoGood = Soomla.Models.EquippableVG.create({
    equipping: Soomla.Models.EquippableVG.EquippingModel.CATEGORY,
    name: "Happi Hippo",
    description: "Happi Hippo description",
    itemId: "happi_hippo",
    purchasableItem: Soomla.Models.PurchaseWithVirtualItem.create({
      pvi_itemId: MUFFIN_CURRENCY_ITEM_ID,
      pvi_amount: 44
    })
  });

  var funkeyMonkeyGood = Soomla.Models.EquippableVG.create({
    equipping: Soomla.Models.EquippableVG.EquippingModel.CATEGORY,
    name: "Funkey Monkey",
    description: "Funkey Monkey description",
    itemId: "funkey_monkey",
    purchasableItem: Soomla.Models.PurchaseWithVirtualItem.create({
      pvi_itemId: MUFFIN_CURRENCY_ITEM_ID,
      pvi_amount: 35
    })
  });


  /** Virtual Categories **/
  var cakes = Soomla.Models.VirtualCategory.create({
    name: "Cakes",
    goods_itemIds: [
      MUFFINCAKE_ITEM_ID,
      PAVLOVA_ITEM_ID,
      "pavlova_10",
      CHOCLATECAKE_ITEM_ID,
      CREAMCUP_ITEM_ID,
      "cream_cup_10"
    ]
  });

  var upgrades = Soomla.Models.VirtualCategory.create({
    name: "Upgrades",
    goods_itemIds: [
      "show_room_0",
      "show_room_1",
      "show_room_2",
      "show_room_3",
      "show_room_4",
      "delivery_vehicle_0",
      "delivery_vehicle_1",
      "delivery_vehicle_2",
      "delivery_vehicle_3",
      "delivery_vehicle_4"
    ]
  });

  var characters = Soomla.Models.VirtualCategory.create({
    name: "Characters",
    goods_itemIds: [
      "fat_cat",
      "happi_hippo",
      "funkey_monkey"
    ]
  });

  /** Google MANAGED Items **/

  var noAdsNoncons = Soomla.Models.NonConsumableItem.create({
    name: "No Ads",
    description: "Test purchase of MANAGED item.",
    itemId: "no_ads",
    purchasableItem: Soomla.Models.PurchaseWithMarket.create({
      marketItem: Soomla.Models.MarketItem.create({
        productId: NO_ADDS_NONCONS_PRODUCT_ID,
        consumable: Soomla.Models.MarketItem.Consumable.NONCONSUMABLE,
        price: 1.99
      })
    })
  });

  var muffinRushAssets = Soomla.IStoreAssets.create({
    categories: [cakes, upgrades, characters],
    currencies: [muffinCurrency],
    currencyPacks: [tenmuffPack, fiftymuffPack, fourhundmuffPack, thousandmuffPack],
    goods: {
      singleUse: [muffincakeGood, pavlovaGood, choclatecakeGood, creamcupGood],
      lifetime: [showRoomGood, deliveryVehicleGood],
      equippable: [fatCatGood, happiHippoGood, funkeyMonkeyGood],
      goodUpgrades: [showRoomGood0, showRoomGood1, showRoomGood2, showRoomGood3, showRoomGood4,
        deliveryVehicleGood0, deliveryVehicleGood1, deliveryVehicleGood2, deliveryVehicleGood3, deliveryVehicleGood4],
      goodPacks: [tenPavlovaGoods, tenCreamcupGoods]
    },
    nonConsumables: [noAdsNoncons],
    version: 1
  });

  return muffinRushAssets;
}

File: src/cocos2dx-store/submodules/android-store/SoomlaAndroidExample/src/com/soomla/example/MuffinRushAssets.java
package com.soomla.example;

import com.soomla.store.IStoreAssets;
import com.soomla.store.domain.*;
import com.soomla.store.domain.virtualCurrencies.*;
import com.soomla.store.domain.virtualGoods.*;
import com.soomla.store.purchaseTypes.*;

import java.util.ArrayList;
import java.util.Arrays;

public class MuffinRushAssets implements IStoreAssets {

    @Override
    public int getVersion() {
        return 0;
    }

    @Override
    public VirtualCurrency[] getCurrencies(){
        return  new VirtualCurrency[] {
                MUFFIN_CURRENCY
        };
    }

    @Override
    public VirtualGood[] getGoods(){
        return new VirtualGood[] {
                MUFFINCAKE_GOOD, PAVLOVA_GOOD,
                CHOCLATECAKE_GOOD, CREAMCUP_GOOD
        };
    }

    @Override
    public VirtualCurrencyPack[] getCurrencyPacks(){
        return new VirtualCurrencyPack[] {
                TENMUFF_PACK, FIFTYMUFF_PACK, FOURHUNDMUFF_PACK, THOUSANDMUFF_PACK
        };
    }

    @Override
    public VirtualCategory[] getCategories() {
        return new VirtualCategory[]{
                GENERAL_CATEGORY
        };
    }

    @Override
    public NonConsumableItem[] getNonConsumableItems() {
        return new NonConsumableItem[]{
                NO_ADDS_NONCONS
        };
    }


    /** Static Final members **/

    public static final String MUFFIN_CURRENCY_ITEM_ID      = "currency_muffin";
    public static final String TENMUFF_PACK_PRODUCT_ID      = "android.test.refunded";
    public static final String FIFTYMUFF_PACK_PRODUCT_ID    = "android.test.canceled";
    public static final String FOURHUNDMUFF_PACK_PRODUCT_ID = "android.test.purchased";
    public static final String THOUSANDMUFF_PACK_PRODUCT_ID = "android.test.item_unavailable";
    public static final String NO_ADDS_NONCONS_PRODUCT_ID   = "no_ads";

    public static final String MUFFINCAKE_ITEM_ID   = "fruit_cake";
    public static final String PAVLOVA_ITEM_ID   = "pavlova";
    public static final String CHOCLATECAKE_ITEM_ID   = "chocolate_cake";
    public static final String CREAMCUP_ITEM_ID   = "cream_cup";

    /** Virtual Currencies **/
    public static final VirtualCurrency MUFFIN_CURRENCY = new VirtualCurrency(
            "Muffins",
            "",
            MUFFIN_CURRENCY_ITEM_ID
    );

    /** Virtual Currency Packs **/

    public static final VirtualCurrencyPack TENMUFF_PACK = new VirtualCurrencyPack(
            "10 Muffins",                                   // name
            "Test refund of an item",                       // description
            "muffins_10",                                   // item id
            10,                                             // number of currencies in the pack
            MUFFIN_CURRENCY_ITEM_ID,                                // the currency associated with this pack
            new PurchaseWithMarket(TENMUFF_PACK_PRODUCT_ID, 0.99));

    public static final VirtualCurrencyPack FIFTYMUFF_PACK = new VirtualCurrencyPack(
            "50 Muffins",                                   // name
            "Test cancellation of an item",                 // description
            "muffins_50",                                   // item id
            50,                                             // number of currencies in the pack
            MUFFIN_CURRENCY_ITEM_ID,                        // the currency associated with this pack
            new PurchaseWithMarket(FIFTYMUFF_PACK_PRODUCT_ID, 1.99));

    public static final VirtualCurrencyPack FOURHUNDMUFF_PACK = new VirtualCurrencyPack(
            "400 Muffins",                                  // name
            "Test purchase of an item",                     // description
            "muffins_400",                                  // item id
            400,                                            // number of currencies in the pack
            MUFFIN_CURRENCY_ITEM_ID,                        // the currency associated with this pack
            new PurchaseWithMarket(FOURHUNDMUFF_PACK_PRODUCT_ID, 4.99));

    public static final VirtualCurrencyPack THOUSANDMUFF_PACK = new VirtualCurrencyPack(
            "1000 Muffins",                                 // name
            "Test item unavailable",                        // description
            "muffins_1000",                                 // item id
            1000,                                           // number of currencies in the pack
            MUFFIN_CURRENCY_ITEM_ID,                        // the currency associated with this pack
            new PurchaseWithMarket(THOUSANDMUFF_PACK_PRODUCT_ID, 8.99));

    /** Virtual Goods **/

    public static final VirtualGood MUFFINCAKE_GOOD = new SingleUseVG(
            "Fruit Cake",                                       // name
            "Customers buy a double portion on each purchase of this cake", // description
            "fruit_cake",                                       // item id
            new PurchaseWithVirtualItem(MUFFIN_CURRENCY_ITEM_ID, 225)); // the way this virtual good is purchased

    public static final VirtualGood PAVLOVA_GOOD = new SingleUseVG(
            "Pavlova",                                          // name
            "Gives customers a sugar rush and they call their friends",    // description
            "pavlova",                                          // item id
            new PurchaseWithVirtualItem(MUFFIN_CURRENCY_ITEM_ID, 175)); // the way this virtual good is purchased

    public static final VirtualGood CHOCLATECAKE_GOOD = new SingleUseVG(
            "Chocolate Cake",                                   // name
            "A classic cake to maximize customer satisfaction", // description
            "chocolate_cake",                                   // item id
            new PurchaseWithVirtualItem(MUFFIN_CURRENCY_ITEM_ID, 250)); // the way this virtual good is purchased


    public static final VirtualGood CREAMCUP_GOOD = new SingleUseVG(
            "Cream Cup",                                        // name
            "Increase bakery reputation with this original pastry",   // description
            "cream_cup",                                        // item id
            new PurchaseWithVirtualItem(MUFFIN_CURRENCY_ITEM_ID, 50));  // the way this virtual good is purchased



    /** Virtual Categories **/

    // The muffin rush theme doesn't support categories, so we just put everything under a general category.
    // In case you're using Soomla' storefront, the order of the virtual goods in this array matters!
    public static final VirtualCategory GENERAL_CATEGORY = new VirtualCategory(
            "General", new ArrayList<String>(Arrays.asList(new String[] { MUFFINCAKE_ITEM_ID, PAVLOVA_ITEM_ID, CHOCLATECAKE_ITEM_ID, CREAMCUP_ITEM_ID }))
    );


    /** Google Non Consumable (MANAGED) Items **/

    public static final NonConsumableItem NO_ADDS_NONCONS  = new NonConsumableItem(
            "No Ads",
            "Test purchase of MANAGED item.",
            "no_ads",
            new PurchaseWithMarket(new MarketItem(NO_ADDS_NONCONS_PRODUCT_ID, MarketItem.Managed.MANAGED , 1.99))
    );

}
